[EDU-3639] feat: add manage object storage guide (#576)
aziontech/docs: src/content/docs/en/pages/guides/object-storage/create-object-storage/index.mdx

---
title: How to create and modify an Azion Object Storage bucket
description: Learn how to create, rename, or delete your buckets from Object Storage using the Azion API.
meta_tags: object storage, storage, cloud, s3, bucket, file storage
namespace: documentation_products_object_storage_bucket
permalink: /documentation/products/guides/how-to-create-and-modify-bucket/
---

**Object Storage** allows you you to create buckets and integrate your object storage with your edge infrastructure. This guide covers how to create, modify, and delete a bucket using the [Azion API](https://api.azion.com).

> Read [How to upload and download objects from a storage bucket](/en/documentation/products/guides/how-to-upload-and-download-objects-from-bucket/) to learn about object operations.

Author: ```suggestion
Read [How to upload and download objects from an Azion Object Storage bucket](/en/documentation/products/guides/how-to-upload-and-download-objects-from-bucket/) to learn about object operations.
```

--header 'Authorization: Token [TOKEN VALUE]' \
--data '{
    "name": "my-bucket-ro",
    "edge-access": "read_only"
}'
```

You should receive the following response:

```json
{
  "state": "executed",
  "data": {
    "name": "my-bucket-ro",
    "edge_access": "read_only"
  }
}
```

Now you can use the bucket name to [upload and download objects](/en/documentation/products/guides/how-to-upload-and-download-objects-from-bucket/) from the bucket.

## Attribute read-write permission to bucket

Run the following `PATCH` request in your terminal, replacing `[TOKEN VALUE]` with your [personal token](/en/documentation/products/guides/personal-tokens/) and `bucket_name` by the name of the bucket to attribute the read-write permission:

```bash
curl --location 'https://api.azion.com/v4/storage/buckets/<bucket_name>' \
--header 'Accept: application/json; version=3' \
--header 'Content-Type: application/json' \
--header 'Authorization: Token [TOKEN VALUE]' \
--data '{
    "edge-access": "read_write"
}'
```

You should receive the following response:

```json
{
  "state": "executed",
  "data": {
    "name": "my-bucket-ro",
    "edge_access": "read_write"
  }
}
```

> 
> You can use the same operation to rename the bucket by sending the new value for the `name` property in the request body.

Author: ```suggestion
```

---
title: How to create and modify an Azion Object Storage bucket
description: Learn how to create, rename, or delete your buckets from Object Storage using the Azion API.
meta_tags: object storage, storage, cloud, s3, bucket, file storage
namespace: documentation_products_object_storage_bucket

> permalink: /documentation/products/guides/how-to-create-and-modify-bucket/

Reviewer: ```suggestion
permalink: /documentation/products/guides/create-and-modify-bucket/
```

---

import Button from '~/components/Button.astro'

> **Object Storage** allows you you to create buckets and integrate your object storage with your edge infrastructure. This guide covers how to create, modify, and delete a bucket using the [Azion API](https://api.azion.com).

Reviewer 2: ```suggestion
**Object Storage** allows you to create buckets and integrate your object storage with your edge infrastructure. This guide covers how to create, modify, and delete a bucket using the [Azion API](https://api.azion.com).
```

> <Button href="/en/documentation/products/store/object-storage/" text="go to Object Storage reference documentation" variant="secondary"></Button>

Reviewer 2: ```suggestion
<Button href="/en/documentation/products/store/object-storage/" text="go to Object Storage reference" variant="secondary"></Button>
```

> Read [How to upload and download objects from an Azion Object Storage bucket](/en/documentation/products/guides/how-to-upload-and-download-objects-from-bucket/) to learn about object operations.

Author: ```suggestion
Read [How to upload and download objects from an Azion Object Storage bucket](/en/documentation/products/guides/upload-and-download-objects-from-bucket/) to learn about object operations.
```

> ## Create a read-only bucket via API

Reviewer: ```suggestion
## Creating a read-only bucket via API
```

> 
> ## Creating a read-only bucket via API

Reviewer 2: ```suggestion

---

## Creating a read-only bucket via API
```

> Run the following `POST` request in your terminal, replacing `[TOKEN VALUE]` with your [personal token](/en/documentation/products/guides/personal-tokens/) and the `name` with the desired bucket name to create a new read-write bucket:

Reviewer: ```suggestion
Run the following `POST` request in your terminal, replacing `[TOKEN VALUE]` with your [personal token](/en/documentation/products/guides/personal-tokens/) and `name` with the desired bucket name to create a new read-write bucket:
```

Reviewer: read-write ou read-only?

```bash
curl --location 'https://api.azion.com/v4/storage/buckets' \
--header 'Accept: application/json; version=3' \
--header 'Content-Type: application/json' \
--header 'Authorization: Token [TOKEN VALUE]' \
--data '{
    "name": "my-bucket-ro",
    "edge-access": "read_only"
}'
```

You should receive the following response:

```json
{
  "state": "executed",
  "data": {
    "name": "my-bucket-ro",
    "edge_access": "read_only"
  }
}
```

> Now you can use the bucket name to [upload and download objects](/en/documentation/products/guides/how-to-upload-and-download-objects-from-bucket/) from the bucket.

Author: ```suggestion
Now you can use the bucket name to [upload and download objects](/en/documentation/products/guides/upload-and-download-objects-from-bucket/) from the bucket.
```

> 
> ## Attributing read-write permission to bucket

Reviewer 2: ```suggestion

---

## Attributing read-write permission to bucket
```

> ## Attribute read-write permission to bucket

Reviewer: ```suggestion
## Attributing read-write permission to bucket
```

> Run the following `PATCH` request in your terminal, replacing `[TOKEN VALUE]` with your [personal token](/en/documentation/products/guides/personal-tokens/) and `bucket_name` by the name of the bucket to attribute the read-write permission:

Reviewer: ```suggestion
Run the following `PATCH` request in your terminal, replacing `[TOKEN VALUE]` with your [personal token](/en/documentation/products/guides/personal-tokens/) and `bucket_name` with the name of the bucket to attribute the read-write permission:
```

```bash
curl --location 'https://api.azion.com/v4/storage/buckets/<bucket_name>' \
--header 'Accept: application/json; version=3' \
--header 'Content-Type: application/json' \
--header 'Authorization: Token [TOKEN VALUE]' \
--data '{
    "edge-access": "read_write"
}'
```

You should receive the following response:

```json
{
  "state": "executed",
  "data": {
    "name": "my-bucket-ro",
    "edge_access": "read_write"
  }
}
```

> 
> ## Removing bucket

Reviewer 2: ```suggestion

---

## Removing bucket
```

> ## Remover bucket

Reviewer: ```suggestion
## Removing bucket
```

> Run the following `DELETE` request in your terminal, replacing `[TOKEN VALUE]` with your [personal token](/en/documentation/products/guides/personal-tokens/) and `bucket_name` by the name of the bucket you want to delete:

Reviewer: ```suggestion
Run the following `DELETE` request in your terminal, replacing `[TOKEN VALUE]` with your [personal token](/en/documentation/products/guides/personal-tokens/) and `bucket_name` with the name of the bucket you want to delete:
```

```bash
curl --location --request DELETE 'https://api.azion.com/v4/storage/buckets/<bucket_name>' \
--header 'Accept: application/json' \
--header 'Authorization: Token [TOKEN VALUE]'
```

> This operation can only be executed if the bucket is empty.

Reviewer 2: ```suggestion
:::note
This operation can only be executed if the bucket is empty.
:::
```
